ProfileManager.js

```diff
@@ -148,34 +211,65 @@ export default class ProfileManager {
     return {
       edv,
       edvConfigZcap: edvZcaps[0],
-      invocationSigner,
       zcaps: edvZcaps,
-      profileAgentId: profileAgent.id
+    };
+  }
+
+  async getCapabilitySetEdv({referenceId, profileAgent}) {
+    const profileId = profileAgent.profile;
+
+    const {kmsClient, invocationSigner, zcaps} = await this.getProfileSigner(
+      {profileAgent});
 
+    // FIXME: can this key be contructed without the search?
+    // the challenge is that the referenceId (zcaps[referenceId]) is a DID
+    // key that includes a hash fragment
+    // e.g. did:key:MOCK_KEY#MOCK_KEY-key-capabilityInvocation
+    const _zcapMapKey = Object.keys(zcaps).find(referenceId => referenceId
+      .endsWith('key-capabilityInvocation'));
+    const zcap = zcaps[_zcapMapKey];
+
+    const keystoreId = _getKeystoreId({zcap});
+    const keystore = await KmsClient.getKeystore({id: keystoreId});
+    const capabilityAgent = new CapabilityAgent(
+      {handle: 'primary', signer: invocationSigner});
+    const keystoreAgent = new KeystoreAgent(
+      {keystore, capabilityAgent, kmsClient});
+    const edvClient = await edvs.get({
+      invocationSigner,
+      keystoreAgent,
+      profileId,
+      referenceId,
+    });
+
+    edvClient.ensureIndex({attribute: 'content.profileAgentId'});
+
+    return {
+      edvClient,
+      invocationSigner,
     };
   }
 
+  // FIXME: this function is getting called frequently and is expensive
```

Comment:
In another life, this library was using an EDV Client cache to mitigate this issue. I believe that code has not been removed. The FIXME can be updated to say to use the cache.